.github/workflows/package_installers.yml

```diff
@@ -33,35 +33,83 @@ jobs:
           # We need to download all tags so that the axion-release-plugin
           # can resolve the most recent version tag.
           fetch-depth: 0
+
       - name: Set up JDK 17
         uses: actions/setup-java@v2
         with:
           # We need a recent version of Java with jpackage included.
           java-version: '17'
           # We use the zulu distribution, which is an OpenJDK distro.
           distribution: 'zulu'
-      - name: Package GUI app installer with Gradle
+
+      # We create a code-signing keychain on MacOS before building and packaging the app, as the
+      # app will be signed as part of the jpackage build phase.
+      - name: "MacOS - Import Certificate: Developer ID Application"
+        if: matrix.os == 'macos-latest'
+        uses: devbotsxyz/import-signing-certificate@main
+        with:
+          # This should be a certificate + private key, exported as a p12 file and base64 encoded as
+          # a GitHub secret.  The certificate should be the:
+          # 'Developer ID Application: The International Data Organization For Transport (BF2U75HN4D)'
+          # certificate, locked with the specified passphrase.
+          certificate-data: ${{ secrets.DEVELOPER_ID_APPLICATION_CERTIFICATE_P12_BASE64 }}
+          certificate-passphrase: ${{ secrets.DEVELOPER_ID_APPLICATION_CERTIFICATE_PASSWORD }}
+          # The resulting keychain will be locked with this separate password.
+          keychain-password: ${{ secrets.KEYCHAIN_PASSWORD }}
+
+      - name: "Package GUI app installer with Gradle"
         uses: gradle/gradle-build-action@v2
         with:
           arguments: jpackage
-        # Per discussion in https://github.com/MobilityData/gtfs-validator/issues/1183, there is a
-        # bug with jpackage in Java 17 that caused Mac apps to be left in an ambiguous signing state
-        # that results in them being reported as 'damaged' when run.  As a temporary workaround, we
-        # unsign the app and repackage as a dmg.  This bug could be fixed in the future by upgrading
-        # to Java 18 or by officially signing the app.
-      - name: Unsign and package Mac OS app
+
+      # On MacOS, we now submit the app for "notarization", where Apple will scan the app for
+      # malware and other issues.  This step can take a few minutes or more; the action will wait
+      # until the report is available.
+      - name: "MacOS - Notarize Release Build"
+        if: matrix.os == 'macos-latest'
+        uses: devbotsxyz/xcode-notarize@v1
```

Comment:
Apple requests that developers limit their # of notarization requests per day (~75/day is guidance).  I don't think we have that many pull requests per day, but I think it might be good to limit this either way.  That could mean restricting to just pushes+releases on the main master branch?

Reply:
> That could mean restricting to just pushes+releases on the main master branch?

I agree with this. I think it's fair solution since we won't be merging ~75 PR/day.